lib/compat/wordpress-6.4/block-hooks.php

```diff
@@ -158,14 +158,28 @@ function gutenberg_insert_hooked_block( $inserted_block, $relative_position, $an
 				array_unshift( $block['innerBlocks'], $inserted_block );
 				// Since WP_Block::render() iterates over `inner_content` (rather than `inner_blocks`)
 				// when rendering blocks, we also need to prepend a value (`null`, to mark a block
-				// location) to that array.
-				array_unshift( $block['innerContent'], null );
+				// location) to that array after HTML content for the inner blocks wrapper.
+				$chunk_index = 0;
+				for ( $index = $chunk_index; $index < count( $block['innerContent'] ); $index++ ) {
+					if ( is_null( $block['innerContent'][ $index ] ) ) {
+						$chunk_index = $index;
+						break;
+					}
+				}
+				array_splice( $block['innerContent'], $chunk_index, 0, array( null ) );
 			} elseif ( 'last_child' === $relative_position ) {
 				array_push( $block['innerBlocks'], $inserted_block );
 				// Since WP_Block::render() iterates over `inner_content` (rather than `inner_blocks`)
-				// when rendering blocks, we also need to prepend a value (`null`, to mark a block
-				// location) to that array.
-				array_push( $block['innerContent'], null );
+				// when rendering blocks, we also need to correctly append a value (`null`, to mark a block
+				// location) to that array before the remaining HTML content for the inner blocks wrapper.
+				$chunk_index = 0;
```

Comment:
We probably want to set 0 when no elements, and the last index when elements are present:

<img width="382" alt="Screenshot 2023-09-13 at 15 31 58" src="https://github.com/WordPress/gutenberg/assets/699132/6947f1ea-e0c7-47fa-b204-c86b076997d3">
<img width="377" alt="Screenshot 2023-09-13 at 15 32 21" src="https://github.com/WordPress/gutenberg/assets/699132/5030a98d-c414-48c1-94fb-ffac922ca42f">

Reply:
I went with a simpler approach `$chunk_index = count( $block['innerContent'] );`. The reason for that is this way, `null` gets appended to the array, rather than prepending. Plus it resolve the original issue.